src/scheduler/pbs_sched.cpp

```diff
@@ -770,20 +654,19 @@ read_sched_cmd(int sock)
 		 * and if we get priority command then just ignore it
 		 * since we are not yet in middle of schedule cycle
 		 */
-		cmd_prio = new_sched_cmd();
-		if (cmd_prio != NULL) {
-			int rc_prio = get_sched_cmd_noblk(sock, cmd_prio);
-			free_sched_cmd(cmd_prio);
-			if (rc_prio == -2) {
-				free_sched_cmd(cmd);
-				return 0;
-			}
-		}
+		int rc_prio = get_sched_cmd_noblk(sock, &cmd_prio);
+		if (rc_prio == -2)
+			return 0;
 	}
-	if (!ds_enqueue(sched_cmds, cmd)) {
-		free_sched_cmd(cmd);
-		return -1;
+
+	if (cmd.cmd != SCH_SCHEDULE_RESTART_CYCLE)  {
+		if (cmd.cmd == SCH_SCHEDULE_AJOB)
+			qrun_list[qrun_list_size++] = cmd;
```

Comment:
SInce one server can get more than one qrun request, you might have to realloc this list array otherwise you will have invalid memory access.

Reply:
As discussed Server only sends once SCH_SCHEDULE_AJOB command at a time even though somebody submitted a qrun with list of jobs. So no need to realloc. At any time scheduler can get at the max the number of qrun requests which is proportional to the number of configured servers. For some more info please look at the following reply.
https://github.com/openpbs/openpbs/pull/2068#discussion_r508242755